### PR TITLE
Update epidemic model to use age brackets

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

This implements a new epidemic projection model that is based on a 2D matrix input of age brackets and SEIR states instead of the flat list we were using previously. I brought in a third-party package for working with N-dimensional arrays to help with this. Outputs were validated against test data from our master model. This also wires up the CurveChart component with the new data context; I had to do it anyway in order to validate that the new model code didn't break anything, so I included that here too.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,10 +3,11 @@
   "scripts": {
     "build": "parcel build index.html && echo && parcel build --target node src/entry-point-static/build.tsx && echo && node dist/build.js && rm -f dist/build.js dist/build.js.map",
     "clean": "rm -rf dist && rm -rf .cache",
-    "dev": "bin/parcel serve index.html",
-    "build": "parcel build index.html && echo && parcel build --target node src/entry-point-static/build.tsx && echo && node dist/build.js && rm -f dist/build.js dist/build.js.map"
+    "dev": "bin/parcel serve index.html"
   },
   "dependencies": {
+    "@types/ndarray": "^1.0.7",
+    "ndarray": "^1.0.19",
```
Comment (szhu) [inline package.json:10]: Can we use the exact version for these?

```suggestion
    "@types/ndarray": "1.0.7",
    "ndarray": "1.0.19",
```

File: package.json [hunk 2]
```diff
@@ -3,10 +3,11 @@
   "scripts": {
     "build": "parcel build index.html && echo && parcel build --target node src/entry-point-static/build.tsx && echo && node dist/build.js && rm -f dist/build.js dist/build.js.map",
     "clean": "rm -rf dist && rm -rf .cache",
-    "dev": "bin/parcel serve index.html",
-    "build": "parcel build index.html && echo && parcel build --target node src/entry-point-static/build.tsx && echo && node dist/build.js && rm -f dist/build.js dist/build.js.map"
```
Comment (szhu) [inline package.json:7]: Thanks for removing this :)

File: src/infection-model/matrixUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// helpers for the ndarray package
```
Comment (szhu) [inline src/infection-model/matrixUtils.tsx:1]: I don't think you need this line -- it's clear that you need it below. Let me know if I'm misunderstanding?

File: package.json (hunk 1 above)
Comment (macfarlandian) [reply package.json:10]: sorry keep forgetting 😬

File: src/impact-dashboard/ImpactDashboard.tsx [hunk 4]
```diff
@@ -21,7 +23,10 @@ const ImpactDashboard: React.FC = () => {
   return (
     <Container>
       <FormContainer>forms</FormContainer>
-      <ChartsContainer>chart goes here</ChartsContainer>
+      <ChartsContainer>
+        chart goes here
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply src/impact-dashboard/ImpactDashboard.tsx:27]: the chart has, indeed, gone there

**Pull request merged**